### ISSUE TITLE
psbt: set global_xpubs (at least for multisig descriptors)

### ISSUE THREAD

**Issue by Sjors** [bitcoin/bitcoin#27583]
### Please describe the feature you'd like to see added.

The `walletcreatefundedpsbt`, `walletprocesspsbt` and `send`* RPC, as well as the send dialog* in the GUI should populate the `PSBT_GLOBAL_XPUB` field (defined in [BIP 174](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki)).

At least when used in a multisig context, e.g. when spending from a `multi()` descriptor.

The Ledger Bitcoin app enforces this as of version 2.1.1., see https://github.com/bitcoin-core/HWI/issues/671

`*` = when used with an external signer

### Is your feature related to a problem, if so please describe it.

Multisig funds (P2WSH and P2SH-P2WSH) are trapped on a Ledger if you update it's Bitcoin app to 2.1.1, unless you use some additional tooling to set this field.

**Comment by Sjors** [comment]
See also this discussion: https://github.com/cryptoadvance/specter-desktop/issues/1886#issuecomment-1246963927

**Comment by BrandonOdiwuor** [comment]
What's the progress with this?

**Comment by BrandonOdiwuor** [comment]
@achow101 what do you think of this?

**Comment by achow101** [comment]
This is more reasonable to do now that legacy wallets are no longer supported.

**Comment by jeanpablojp** [comment]
I got here reading the PSBT code while doing a review, and went to check whether this still applies. From what I checked today, on a `wsh(multi(2,...))` wallet whose descriptors already carry origin and extended key, `walletcreatefundedpsbt`, `walletprocesspsbt` and `send` all return an empty `global_xpubs`. The field already serializes, deserializes and merges when combining PSBTs, it's just that nothing populates it.

@achow101, this is assigned to you, are you still planning to pick it up? If so, I'll leave it with you. If it's stalled, I can prepare a patch.

**Comment by Sjors** [comment]
@jeanpablojp I would not worry about assignment. Feel free to work on this.

That said, I'm not sure if my initial motivation still matters, since Ledger uses BIP388 policies now. See also https://github.com/Sjors/bitcoin/pull/91.

**Comment by jeanpablojp** [comment]
@Sjors I went to check. For multisig, the policies are built by HWI from the PSBT's global xpubs, so the field still feeds that path. I saw in your bitcoin-core/HWI#794 that you add a registration path, and that you leave the Trezor out of it, with `parse_multisig` still reading `psbt.xpub`. Without registration the information has to come inside the PSBT, and Specter fills it in itself after getting the PSBT from Core. And even with registration everywhere, whoever signs with just the file, without a wallet loaded, has no policy and no registration to pass.

If it makes any sense to go ahead, do we fill it always, or only when the descriptor is multisig?

**Comment by Sjors** [comment]
@jeanpablojp so it sounds like we still need global xpubs for at least Trezor?

**Comment by jeanpablojp** [comment]
@Sjors Yes, and from what I saw it's not only the Trezor, in your PR `digitalbitbox.py`, `keepkey.py` and `ledger.py` with the legacy app raise the same exception.
